pkg/config/config.go

```diff
@@ -63,6 +63,10 @@ const (
 	// disableHTTP2AnnotationKey is the annotation key attached to a Knative Domain Mapping
 	// to indicate that http2 should not be enabled for it.
 	disableHTTP2AnnotationKey = "kourier.knative.dev/disable-http2"
+
+	// ServingNamespaceEnv is an env variable specifying where the serving is deployed.
+	// e.g. OpenShift deploys Kourier in different namespace so `system.Namespace()` does not work.
```

Comment:
Afaik this holds for backwards compatibility reasons. Is there anything that stops us from moving everything to `serving` ns?

Reply:
It is possible but it has a lot of risks to move the gateway into serving ns.